src/mongo_logger.ts

```diff
@@ -421,13 +422,44 @@ export function stringifyWithMaxLen(
 ): string {
   let strToTruncate = '';
 
+  let currentLength = 0;
+  const maxDocumentLengthEnsurer = function maxDocumentLengthEnsurer(key: string, value: any) {
+    if (currentLength >= maxDocumentLength) {
+      return undefined;
+    }
+
+    currentLength += key.length + 4;
+
+    if (typeof value === 'string') {
+      currentLength += value.length;
+    } else if (typeof value === 'number' || typeof value === 'bigint') {
+      currentLength += 20;
```

Comment:
where are these lengths coming from?  wouldn't the length of numbers be dependent on the actual number being stringified (ex: `0` would be 1, but `1.1234` would be 6?)

Reply:
Yes correct they would depend on the value. I can improve this to calculate the number of digits from the value. 

Generally, the lengths here are approximations when we can't determine the length without actually stringifying the value in an attempt to avoid paying that potentially expensive cost. With a better implementation, we should always get a bit over maxLen returned to us and then slicing ensures we actually hit the target.